### PR TITLE
[Upgrade] Move to NSs, autoloading with PSR-4 and dependency management

### LINKED ISSUE #8: Update Client to Adhere to FIG Standards
http://www.php-fig.org/

Should adhere to PSR-1 and PSR-2. Adherance to PSR-0 would also be good, but will likely require substantial updating to accomplish.

### PR DESCRIPTION
**Pull request by phansys**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   |
| Fixed tickets | #8
| License       | MIT
| Doc PR        |

Fixes #8.

* Moved class names to namespaces respecting [PSR-4](http://www.php-fig.org/psr/psr-4/)
  autoload standard.
* Added [Composer](https://getcomposer.org/) as dependency manager

In order to allow this library to be installed with Composer, it must be submitted
in [Packagist](https://packagist.org/packages/submit) after this PR is merged.

### CODE REVIEW

**Comment by underscorephil** [comment]
Thanks for the PR!

Looking at this.

**Comment by phansys** [comment]
Thank you @underscorephil. Please let me know if you have any doubt or comment about these changes.

File: src/XmlRpcClient.php [hunk 1]
```diff
@@ -92,7 +86,7 @@ class Softlayer_XmlrpcClient
      *
      * @var string
      */
-    const API_BASE_URL = SoftLayer_XmlrpcClient::API_PUBLIC_ENDPOINT;
+    const API_BASE_URL = XmlRpcClient::API_PUBLIC_ENDPOINT;
```
Comment (underscorephil) [inline src/XmlRpcClient.php]: Should this be self:: here instead of XmlRpcClinet::?

**Comment by underscorephil** [comment]
@phansys ready to merge after I get your response on the line note

**Comment by phansys** [comment]
Sorry for the delay @underscorephil. PR updated and rebased.
I left the class name instead `self` keyword before because I was affraid that syntax were not supported in earlier PHP versions, but it seems to be supported from +5.2.4, so isn't an issue there: https://3v4l.org/vGqD7.

Thank you.

**Comment by phansys** [comment]
Nice to find this [library in Packagist](https://packagist.org/packages/softlayer/softlayer-api-php-client)!
It would be nice if you could [tag a release from `0.x` branch](https://github.com/softlayer/softlayer-api-php-client/commit/fb25f1ef46483ce557b65391281e52c4596d1243) in order to allow people looking for a earlier stable release to use it.

**Pull request merged**